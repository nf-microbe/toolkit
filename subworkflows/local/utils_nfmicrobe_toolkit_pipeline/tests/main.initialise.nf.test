nextflow_workflow {

    name "Test Subworkflow PIPELINE_INITIALISATION"
    script "../main.nf"
    workflow "PIPELINE_INITIALISATION"

    tag "subworkflows"
    tag "subworkflows/utils_nfmicrobe_toolkit_pipeline"

    test("samplesheet only") {
        when {
            params {
                input = "${projectDir}/assets/test-datasets/samplesheets/test_samplesheet.csv"
            }
            workflow {
                """
                input[0] = false
                input[1] = false
                input[2] = false
                input[3] = false
                input[4] = []
                input[5] = null
                input[6] = "${projectDir}/assets/test-datasets/samplesheets/test_samplesheet.csv"
                input[7] = []
                input[8] = []
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("fastq only") {
        when {
            workflow {
                """
                input[0] = false
                input[1] = false
                input[2] = false
                input[3] = false
                input[4] = []
                input[5] = null
                input[6] = []
                input[7] = ["https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz", "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz", "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test2_1.fastq.gz"]
                input[8] = []
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("fasta only") {
        when {
            workflow {
                """
                input[0] = false
                input[1] = false
                input[2] = false
                input[3] = false
                input[4] = []
                input[5] = null
                input[6] = []
                input[7] = []
                input[8] = ["https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz", "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/proteome.fasta.gz"]
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("all inputs") {
        when {
            params {
                input = "${projectDir}/assets/test-datasets/samplesheets/test_samplesheet.csv"
            }
            workflow {
                """
                input[0] = false
                input[1] = false
                input[2] = false
                input[3] = false
                input[4] = []
                input[5] = null
                input[6] = "${projectDir}/assets/test-datasets/samplesheets/test_samplesheet.csv"
                input[7] = ["https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz", "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz", "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test2_1.fastq.gz"]
                input[8] = ["https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz", "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/proteome.fasta.gz"]
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
