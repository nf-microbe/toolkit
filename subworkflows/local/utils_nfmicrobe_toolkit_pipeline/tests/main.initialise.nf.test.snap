{
    "all inputs": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "paired_end_no_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "paired_end_no_fasta",
                            "run": "2",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "paired_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_no_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_no_fasta",
                            "run": "2",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test",
                            "run": 0,
                            "group": "test",
                            "single_end": false
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec",
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ],
                    [
                        {
                            "id": "test2",
                            "run": 0,
                            "group": "test2",
                            "single_end": true
                        },
                        [
                            "test2.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                        ]
                    ],
                    [
                        {
                            "id": "test_1",
                            "run": 0,
                            "group": "test_1",
                            "single_end": true
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                        ]
                    ],
                    [
                        {
                            "id": "test_2",
                            "run": 0,
                            "group": "test_2",
                            "single_end": true
                        },
                        [
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652",
                            "run": 0,
                            "group": "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652"
                        },
                        "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652"
                    ],
                    [
                        {
                            "id": "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6",
                            "run": 0,
                            "group": "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                        },
                        "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                    ],
                    [
                        {
                            "id": "fasta_no_run_or_group",
                            "run": "0",
                            "group": "fasta_no_run_or_group",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "fasta_only",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "paired_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "single_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ]
                ],
                "2": [
                    
                ],
                "fastas": [
                    [
                        {
                            "id": "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652",
                            "run": 0,
                            "group": "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652"
                        },
                        "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652"
                    ],
                    [
                        {
                            "id": "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6",
                            "run": 0,
                            "group": "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                        },
                        "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                    ],
                    [
                        {
                            "id": "fasta_no_run_or_group",
                            "run": "0",
                            "group": "fasta_no_run_or_group",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "fasta_only",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "paired_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "single_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ]
                ],
                "fastqs": [
                    [
                        {
                            "id": "paired_end_no_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "paired_end_no_fasta",
                            "run": "2",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "paired_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_no_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_no_fasta",
                            "run": "2",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "test",
                            "run": 0,
                            "group": "test",
                            "single_end": false
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec",
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ],
                    [
                        {
                            "id": "test2",
                            "run": 0,
                            "group": "test2",
                            "single_end": true
                        },
                        [
                            "test2.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                        ]
                    ],
                    [
                        {
                            "id": "test_1",
                            "run": 0,
                            "group": "test_1",
                            "single_end": true
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                        ]
                    ],
                    [
                        {
                            "id": "test_2",
                            "run": 0,
                            "group": "test_2",
                            "single_end": true
                        },
                        [
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ]
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-03T06:58:57.16183429"
    },
    "fastq only": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "run": 0,
                            "group": "test",
                            "single_end": false
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec",
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ],
                    [
                        {
                            "id": "test2",
                            "run": 0,
                            "group": "test2",
                            "single_end": true
                        },
                        [
                            "test2.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                        ]
                    ],
                    [
                        {
                            "id": "test_1",
                            "run": 0,
                            "group": "test_1",
                            "single_end": true
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                        ]
                    ],
                    [
                        {
                            "id": "test_2",
                            "run": 0,
                            "group": "test_2",
                            "single_end": true
                        },
                        [
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "fastas": [
                    
                ],
                "fastqs": [
                    [
                        {
                            "id": "test",
                            "run": 0,
                            "group": "test",
                            "single_end": false
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec",
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ],
                    [
                        {
                            "id": "test2",
                            "run": 0,
                            "group": "test2",
                            "single_end": true
                        },
                        [
                            "test2.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                        ]
                    ],
                    [
                        {
                            "id": "test_1",
                            "run": 0,
                            "group": "test_1",
                            "single_end": true
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                        ]
                    ],
                    [
                        {
                            "id": "test_2",
                            "run": 0,
                            "group": "test_2",
                            "single_end": true
                        },
                        [
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ]
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-03T06:55:26.238983025"
    },
    "samplesheet only": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "paired_end_no_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "paired_end_no_fasta",
                            "run": "2",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "paired_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_no_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_no_fasta",
                            "run": "2",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "fasta_no_run_or_group",
                            "run": "0",
                            "group": "fasta_no_run_or_group",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "fasta_only",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "paired_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "single_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ]
                ],
                "2": [
                    
                ],
                "fastas": [
                    [
                        {
                            "id": "fasta_no_run_or_group",
                            "run": "0",
                            "group": "fasta_no_run_or_group",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "fasta_only",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "paired_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ],
                    [
                        {
                            "id": "single_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        "/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta.gz"
                    ]
                ],
                "fastqs": [
                    [
                        {
                            "id": "paired_end_no_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "paired_end_no_fasta",
                            "run": "2",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "paired_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz",
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_no_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_no_fasta",
                            "run": "2",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "id": "single_end_w_fasta",
                            "run": "1",
                            "group": "group1",
                            "single_end": true
                        },
                        [
                            "/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz"
                        ]
                    ]
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T22:35:26.018366559"
    },
    "fasta only": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652",
                            "run": 0,
                            "group": "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652"
                        },
                        "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652"
                    ],
                    [
                        {
                            "id": "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6",
                            "run": 0,
                            "group": "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                        },
                        "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                    ]
                ],
                "2": [
                    
                ],
                "fastas": [
                    [
                        {
                            "id": "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652",
                            "run": 0,
                            "group": "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652"
                        },
                        "contigs.fasta.gz:md5,c96caedf8c949a2b871b83152d99c652"
                    ],
                    [
                        {
                            "id": "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6",
                            "run": 0,
                            "group": "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                        },
                        "genome.fasta.gz:md5,6e9fe4042a72f2345f644f239272b7e6"
                    ]
                ],
                "fastqs": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-03T06:58:35.604631851"
    }
}